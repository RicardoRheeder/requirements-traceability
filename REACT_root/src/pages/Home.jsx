import React, { Component } from 'react'

import { ReactFlow_Test, CustomNodeFlow } from './'

// Testing REACT-flow-renderer: https://dev.to/vaibhavkhulbe/make-interactive-node-based-graphs-with-react-flow-102d

export default class Home extends Component {
    constructor(props) {
        super(props);
        this.state = { isBasic: true };
    }

    render() {
        return (
            <div>
                Node REACT-flow prototyping
                <br />
                *Note: Cannot have more than one graph render at a time*
                <br />
                <button onClick={() => { this.setState({ isBasic: !this.state.isBasic }) }}>
                    Switch Graph
                </button>
                {this.state.isBasic ? <ReactFlow_Test /> : <CustomNodeFlow />}
            </div>
        )
    }
}
