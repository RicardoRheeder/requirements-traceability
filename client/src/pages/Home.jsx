import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { useEffect } from 'react'
import { useAuth0 } from '@auth0/auth0-react'
import { fetchUserInfoAsync } from '../redux/stores/user/actions'
import { fetchUserDocsAsync } from '../redux/stores/document/actions'
import { LeftContainer } from '../components'

function Home() {
  // getting the current user who is signed in
  const { user } = useAuth0()
  const dispatch = useDispatch()

  // getting info and errorMessage from initial state
  const userInfo = useSelector((state) => state.user.info)
  const errorMessage = useSelector((state) => state.user.errorMessage)

  // use effect to fetch the user info when the component mounts
  useEffect(() => {
    if (user) {
      // dispatching async call with the user as a parameter
      dispatch(fetchUserInfoAsync(user))
      dispatch(fetchUserDocsAsync(user))
    }
  }, [fetchUserInfoAsync, fetchUserDocsAsync])

  // console.log(user)
  // console.log(userInfo)
  // console.log(errorMessage)

  return (
    <div className="home-root styled-background-blue">
      <div className="left-container">
        <LeftContainer />
      </div>
      <div className="center-container">
        <div className="home-header">
          <img
            className="home-logo-banner"
            src="/assets/images/Doc_Tracer_Logo_2.png"
          ></img>
        </div>
        <div className="home-subheader">Recent Documents</div>
      </div>
      <div className="right-container">Notifications</div>
    </div>
  )
}
export default Home
