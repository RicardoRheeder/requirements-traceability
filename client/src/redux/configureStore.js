import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

// restricting extension in production using process.env.NODE_ENV which is set 'production'
import { composeWithDevTools } from 'redux-devtools-extension/logOnlyInProduction';

import store from './stores';

const enhancer = (process.env.NODE_ENV == 'development')
               ? composeWithDevTools(applyMiddleware(thunk))
               : applyMiddleware(thunk);

export default () => createStore(store, enhancer);
