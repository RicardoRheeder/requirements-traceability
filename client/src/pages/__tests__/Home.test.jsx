import React from "react"
import Enzyme, { shallow, mount} from "enzyme"
import Home from "../Home"
import Adapter from "enzyme-adapter-react-16"
import { Provider } from "react-redux";
import configureStore from "redux-mock-store"


const mockStore = configureStore([])
Enzyme.configure({adapter: new Adapter() });

describe("Homepage", () => {
    let store;
    let homepage;
    beforeEach(() => {
        store = mockStore({});
    });
    test("renders", () => {
        const wrapper = shallow(<Provider store = {store}><Home/></Provider>);
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
});