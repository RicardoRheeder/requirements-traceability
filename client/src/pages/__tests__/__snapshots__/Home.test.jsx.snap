// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage renders 1`] = `ShallowWrapper {}`;

exports[`Homepage renders with children 1`] = `ReactWrapper {}`;
